### PR TITLE
Sample functions

### PR DESCRIPTION
**Pull request by TibbTobb**
New regex added which matches all backtrace outputs.
Print address is turned off and restored to make parsing output easier.
Replaced while loop with for loop.

### CODE REVIEW

File: sample_functions/sample_functions.py [hunk 1]
```diff
@@ -51,29 +51,27 @@ def invoke(arg, from_tty):
...
-            if m is None:
-                continue
-            function = m.group(1)
-
-            functions[function] += 1
+            frame = gdb.newest_frame()
+            # Create list of functions in the backtrace
+            trace_functions = []
+            while frame is not None:
+                trace_functions.append(frame.name())
```
Comment (barisione) [inline sample_functions/sample_functions.py]: The name could be `None`. In this case maybe you could append it's pc (`str(frame.pc())`).

File: sample_functions/sample_functions.py [hunk 2]
```diff
@@ -51,29 +51,27 @@ def invoke(arg, from_tty):
         end_bbcount = int(args[1])
         interval = int(args[2])
 
-        current_bbcount = start_bbcount
+        function_p = re.compile(r'#[0-9]+  ([^\s]+) \(.*\) (at|from) .*')
```
Comment (barisione) [inline sample_functions/sample_functions.py]: This is not needed any more.

Comment (TibbTobb) [reply sample_functions/sample_functions.py]: Done

File: sample_functions/sample_functions.py (hunk 1 above)
Comment (TibbTobb) [reply sample_functions/sample_functions.py]: Done

**Pull request merged**